jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -151,6 +152,16 @@ default int capacity()
         return getByteBuffer().capacity();
     }
 
+    default int space()
+    {
+        return capacity() - remaining();
+    }
+
+    default boolean isFull()
+    {
+        return remaining() == capacity();
```

Comment:
Shouldn't that be `return remaining() == 0`?

Reply:
No, as we are in flush mode and remaining is the size of valid bytes, not available empty bytes.